# Background and Context:

Twitter possesses 330 million monthly active users, which allows businesses to reach a broad population and connect with customers without intermediaries. On the other hand, there’s so much information that it’s difficult for brands to quickly detect negative social mentions that could harm their business.

That's why sentiment analysis/classification, which involves monitoring emotions in conversations on social media platforms, has become a key strategy in social media marketing.


Listening to how customers feel about the product/service on Twitter allows companies to understand their audience, keep on top of what’s being said about their brand and their competitors, and discover new trends in the industry.

 

# Data Description:

A sentiment analysis job about the problems of each major U.S. airline. Twitter data was scraped from February of 2015 and contributors were asked to first classify positive, negative, and neutral tweets, followed by categorizing negative reasons (such as "late flight" or "rude service").


# Dataset:

The dataset has the following columns:
- tweet_id                                                           
- airline_sentiment                                               
- airline_sentiment_confidence                               
- negativereason                                                   
- negativereason_confidence                                    
- airline                                                                    
- airline_sentiment_gold                                              
- name     
- negativereason_gold 
- retweet_count
- text
- tweet_coord
- tweet_created
- tweet_location 
- user_timezone

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
pd.set_option('display.max_column', None)
sns.set()

import warnings
warnings.filterwarnings('ignore')

# Import Data

In [2]:
airline = pd.read_csv('Tweets.csv')

In [3]:
df = airline.copy()

In [4]:
df.head()

tweet_id airline_sentiment  airline_sentiment_confidence  \
0  570306133677760513           neutral                        1.0000   
1  570301130888122368          positive                        0.3486   
2  570301083672813571           neutral                        0.6837   
3  570301031407624196          negative                        1.0000   
4  570300817074462722          negative                        1.0000   

  negativereason  negativereason_confidence         airline  \
0            NaN                        NaN  Virgin America   
1            NaN                     0.0000  Virgin America   
2            NaN                        NaN  Virgin America   
3     Bad Flight                     0.7033  Virgin America   
4     Can't Tell                     1.0000  Virgin America   

  airline_sentiment_gold        name negativereason_gold  retweet_count  \
0                    NaN     cairdin                 NaN              0   
1                    NaN    jnardino                 NaN              0   
2                    NaN  yvonnalynn                 NaN              0   
3                    NaN    jnardino                 NaN              0   
4                    NaN    jnardino                 NaN              0   

                                                text tweet_coord  \
0                @VirginAmerica What @dhepburn said.         NaN   
1  @VirginAmerica plus you've added commercials t...         NaN   
2  @VirginAmerica I didn't today... Must mean I n...         NaN   
3  @VirginAmerica it's really aggressive to blast...         NaN   
4  @VirginAmerica and it's a really big bad thing...         NaN   

               tweet_created tweet_location               user_timezone  
0  2015-02-24 11:35:52 -0800            NaN  Eastern Time (US & Canada)  
1  2015-02-24 11:15:59 -0800            NaN  Pacific Time (US & Canada)  
2  2015-02-24 11:15:48 -0800      Lets Play  Central Time (US & Canada)  
3  2015-02-24 11:15:36 -0800            NaN  Pacific Time (US & Canada)  
4  2015-02-24 11:14:45 -0800            NaN  Pacific Time (US & Canada)

In [5]:
df.shape

(14640, 15)

- The dataset consists of 14640 rows and 15 columns.
- Missing values present in serveral columns.
- Features with ambiguous name like `airline_sentiment_gold` and `tweet_coord` may need further investigation to understand what they really mean.

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14640 entries, 0 to 14639
Data columns (total 15 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   tweet_id                      14640 non-null  int64  
 1   airline_sentiment             14640 non-null  object 
 2   airline_sentiment_confidence  14640 non-null  float64
 3   negativereason                9178 non-null   object 
 4   negativereason_confidence     10522 non-null  float64
 5   airline                       14640 non-null  object 
 6   airline_sentiment_gold        40 non-null     object 
 7   name                          14640 non-null  object 
 8   negativereason_gold           32 non-null     object 
 9   retweet_count                 14640 non-null  int64  
 10  text                          14640 non-null  object 
 11  tweet_coord                   1019 non-null   object 
 12  tweet_created                 14640 non-null  object 
 13  t

- Most columns are objects. Only `tweet_id`, `airline_sentiment_confidence`, `negativereason_confidence` and `retweet_count` are numbers.

## Check Missing Values

In [7]:
# show columns with missing values and the proportion of missing values
def check_missing_values():
    miss_df = pd.DataFrame(df.isnull().sum(), columns = ['# of missing values'])
    miss_df["% of missing values"] = np.round(miss_df['# of missing values']/ df.shape[0] * 100, 2)
    return  miss_df[miss_df["% of missing values"] > 0]

In [8]:
check_missing_values()

# of missing values  % of missing values
negativereason                            5462                37.31
negativereason_confidence                 4118                28.13
airline_sentiment_gold                   14600                99.73
negativereason_gold                      14608                99.78
tweet_coord                              13621                93.04
tweet_location                            4733                32.33
user_timezone                             4820                32.92

- Missing values in `airline_sentiment_gold`, `negativereason_gold`, `tweet_coord` account for 90+% of the total.
- Missing values in `negativereason`, `negativereason_confidence`, `tweet_location` and `user_timezone` account for about 30% of the total.

### Drop `airline_sentiment_gold`

In [9]:
df[df.airline_sentiment_gold.notnull()]  # check airline_sentiment_gold with values

tweet_id airline_sentiment  airline_sentiment_confidence  \
1286   569851578276048896          negative                        1.0000   
2017   569473998519578624          negative                        1.0000   
2135   569343661063823360           neutral                        1.0000   
2201   569318505755103232          negative                        1.0000   
2383   569235954789015552          negative                        0.9286   
3005   568802013196242944          negative                        1.0000   
3149   568637541513089024          negative                        0.9220   
4206   567778009013178368          negative                        1.0000   
4525   570066283233972224          positive                        0.9657   
5183   569273733334077440          negative                        1.0000   
5341   569179898532761600          negative                        1.0000   
5749   568752276040495104           neutral                        0.9160   
5810   568581908075933696          negative                        0.9658   
6530   567800574051151872          negative                        0.9277   
6632   567751489754382336          negative                        0.9594   
6728   567717985092395008           neutral                        0.9633   
6758   570295207763316736          negative                        1.0000   
8096   568797120410619904          negative                        0.9278   
8207   568606685230555136          negative                        0.7723   
8536   568182544124014592          negative                        1.0000   
8739   567885370064506880          negative                        1.0000   
9493   569907782108839938          negative                        0.9670   
9536   569887533267611648          negative                        0.8563   
11053  568561924985782272          positive                        1.0000   
12024  570264948548313088          positive                        1.0000   
12025  570139793608175616          negative                        1.0000   
12026  569944281512685570          negative                        1.0000   
12027  569842758967386112          negative                        1.0000   
12028  569699455919726593          negative                        1.0000   
12029  569680231012773888          negative                        1.0000   
12030  569622568459636736          negative                        1.0000   
12031  569621879633391616          negative                        1.0000   
12032  569601363799359488          negative                        0.9553   
12033  569600137296633856          positive                        0.9236   
12034  569047438880841728          negative                        1.0000   
12035  568824537338417154          negative                        1.0000   
12036  568551906634797056          positive                        0.6242   
12037  568265091226800130          negative                        0.9240   
12038  570294451261874177          negative                        1.0000   
12039  568757671819661314          negative                        0.7991   

                    negativereason  negativereason_confidence     airline  \
1286                   Late Flight                     0.7684      United   
2017                   Late Flight                     0.6048      United   
2135                           NaN                        NaN      United   
2201                   Late Flight                     0.6478      United   
2383        Customer Service Issue                     0.9286      United   
3005                    Can't Tell                     0.6870      United   
3149        Customer Service Issue                     0.4513      United   
4206              Cancelled Flight                     1.0000      United   
4525                           NaN                     0.0000   Southwest   
5183        Customer Service Issue                     0.7715   Southwest   
5341        

In [10]:
np.all(df.loc[df.airline_sentiment_gold.notnull(), 'airline_sentiment_gold'] == df.loc[df.airline_sentiment_gold.notnull(), 'airline_sentiment'])

True

- `airline_sentiment_gold` is the replicate of `airline_sentiment`. We could drop `airline_sentiment_gold`.

In [11]:
df.drop(columns = ['airline_sentiment_gold'], inplace = True)

### Treat `negativereason` and `negativereason_gold`

In [12]:
valid_negativereasons = df[df.negativereason_gold.notnull()]  # check negativereason_gold with values
valid_negativereasons

tweet_id airline_sentiment  airline_sentiment_confidence  \
1286   569851578276048896          negative                        1.0000   
2017   569473998519578624          negative                        1.0000   
2201   569318505755103232          negative                        1.0000   
2383   569235954789015552          negative                        0.9286   
3005   568802013196242944          negative                        1.0000   
3149   568637541513089024          negative                        0.9220   
4206   567778009013178368          negative                        1.0000   
5183   569273733334077440          negative                        1.0000   
5341   569179898532761600          negative                        1.0000   
5810   568581908075933696          negative                        0.9658   
6530   567800574051151872          negative                        0.9277   
6632   567751489754382336          negative                        0.9594   
6758   570295207763316736          negative                        1.0000   
8096   568797120410619904          negative                        0.9278   
8207   568606685230555136          negative                        0.7723   
8536   568182544124014592          negative                        1.0000   
8739   567885370064506880          negative                        1.0000   
9493   569907782108839938          negative                        0.9670   
9536   569887533267611648          negative                        0.8563   
12025  570139793608175616          negative                        1.0000   
12026  569944281512685570          negative                        1.0000   
12027  569842758967386112          negative                        1.0000   
12028  569699455919726593          negative                        1.0000   
12029  569680231012773888          negative                        1.0000   
12030  569622568459636736          negative                        1.0000   
12031  569621879633391616          negative                        1.0000   
12032  569601363799359488          negative                        0.9553   
12034  569047438880841728          negative                        1.0000   
12035  568824537338417154          negative                        1.0000   
12037  568265091226800130          negative                        0.9240   
12038  570294451261874177          negative                        1.0000   
12039  568757671819661314          negative                        0.7991   

                    negativereason  negativereason_confidence     airline  \
1286                   Late Flight                     0.7684      United   
2017                   Late Flight                     0.6048      United   
2201                   Late Flight                     0.6478      United   
2383        Customer Service Issue                     0.9286      United   
3005                    Can't Tell                     0.6870      United   
3149        Customer Service Issue                     0.4513      United   
4206              Cancelled Flight                     1.0000      United   
5183        Customer Service Issue                     0.7715   Southwest   
5341        Customer Service Issue                     0.9256   Southwest   
5810                    Bad Flight                     0.6623   Southwest   
6530        Customer Service Issue                     0.4821   Southwest   
6632        Customer Service Issue                     0.9594   Southwest   
6758              Cancelled Flight                     1.0000       Delta   
8096                   Late Flight                     0.5386       Delta   
8207                    Can't Tell                     0.7310       Delta   
8536                  Lost Luggage                     0.8436       Delta   
8739              Cancelled Flight                     1.0000       Delta   
9493                    Can't Tell                     0.8042  US Airways   
9536        

In [13]:
for i, r in valid_negativereasons.iterrows():
    if r['negativereason'] not in r['negativereason_gold']:
        print(i)

12038


- In most cases, elements in `negativereason` are already included in `negativereason_gold`. Only at row 12038, the two don't overlap. 

In [14]:
df[df.index == 12038]

tweet_id airline_sentiment  airline_sentiment_confidence  \
12038  570294451261874177          negative                           1.0   

      negativereason  negativereason_confidence airline          name  \
12038      longlines                     0.4686   Delta  DetroitRonin   

          negativereason_gold  retweet_count  \
12038  Customer Service Issue              0   

                                                    text tweet_coord  \
12038  @DeltaAssist now at 57 minutes waiting on Silv...         NaN   

                   tweet_created tweet_location               user_timezone  
12038  2015-02-24 10:49:27 -0800       Thataway  Eastern Time (US & Canada)

In [15]:
df.negativereason.value_counts()

Customer Service Issue         2910
Late Flight                    1665
Can't Tell                     1190
Cancelled Flight                847
Lost Luggage                    724
Bad Flight                      580
Flight Booking Problems         529
Flight Attendant Complaints     481
longlines                       178
Damaged Luggage                  74
Name: negativereason, dtype: int64

In [16]:
df.negativereason_gold.value_counts()

Customer Service Issue                      12
Late Flight                                  4
Cancelled Flight                             3
Can't Tell                                   3
Cancelled Flight\nCustomer Service Issue     2
Bad Flight                                   1
Customer Service Issue\nLost Luggage         1
Late Flight\nFlight Attendant Complaints     1
Flight Attendant Complaints                  1
Lost Luggage\nDamaged Luggage                1
Customer Service Issue\nCan't Tell           1
Late Flight\nLost Luggage                    1
Late Flight\nCancelled Flight                1
Name: negativereason_gold, dtype: int64

- Sometimes there're more than one issue that make customers unhappy. So there will be more than one negative reason. These elements in `negativereason_gold` are the combinations of the elements in `negativereason`.

**Merge two columns into one, stored in `negativereason`. Drop `negativereason_gold`.**

In [17]:
valid_negativereasons.loc[12038, 'negativereason_gold'] += '\n' + valid_negativereasons.loc[12038, 'negativereason']

for i, r in valid_negativereasons.iterrows():
    df.loc[i, 'negativereason'] = r['negativereason_gold']
    
df[df.index.isin(valid_negativereasons.index)]

tweet_id airline_sentiment  airline_sentiment_confidence  \
1286   569851578276048896          negative                        1.0000   
2017   569473998519578624          negative                        1.0000   
2201   569318505755103232          negative                        1.0000   
2383   569235954789015552          negative                        0.9286   
3005   568802013196242944          negative                        1.0000   
3149   568637541513089024          negative                        0.9220   
4206   567778009013178368          negative                        1.0000   
5183   569273733334077440          negative                        1.0000   
5341   569179898532761600          negative                        1.0000   
5810   568581908075933696          negative                        0.9658   
6530   567800574051151872          negative                        0.9277   
6632   567751489754382336          negative                        0.9594   
6758   570295207763316736          negative                        1.0000   
8096   568797120410619904          negative                        0.9278   
8207   568606685230555136          negative                        0.7723   
8536   568182544124014592          negative                        1.0000   
8739   567885370064506880          negative                        1.0000   
9493   569907782108839938          negative                        0.9670   
9536   569887533267611648          negative                        0.8563   
12025  570139793608175616          negative                        1.0000   
12026  569944281512685570          negative                        1.0000   
12027  569842758967386112          negative                        1.0000   
12028  569699455919726593          negative                        1.0000   
12029  569680231012773888          negative                        1.0000   
12030  569622568459636736          negative                        1.0000   
12031  569621879633391616          negative                        1.0000   
12032  569601363799359488          negative                        0.9553   
12034  569047438880841728          negative                        1.0000   
12035  568824537338417154          negative                        1.0000   
12037  568265091226800130          negative                        0.9240   
12038  570294451261874177          negative                        1.0000   
12039  568757671819661314          negative                        0.7991   

                                 negativereason  negativereason_confidence  \
1286   Late Flight\nFlight Attendant Complaints                     0.7684   
2017                  Late Flight\nLost Luggage                     0.6048   
2201                                Late Flight                     0.6478   
2383                     Customer Service Issue                     0.9286   
3005                                 Can't Tell                     0.6870   
3149   Cancelled Flight\nCustomer Service Issue                     0.4513   
4206                           Cancelled Flight                     1.0000   
5183                     Customer Service Issue                     0.7715   
5341                     Customer Service Issue                     0.9256   
5810                                 Bad Flight                     0.6623   
6530   Cancelled Flight\nCustomer Service Issue                     0.4821   
6632                     Customer Service Issue                     0.9594   
6758                           Cancelled Flight                     1.0000   
8096                                Late Flight                     0.5386   
8207                                 Can't Tell                     0.7310   
8536              Lost Luggage\nDamaged Luggage                     0.8436   
8739                           Cancelled Flight                     1.0000   
9493                                 Can't Tell                     0.8

In [18]:
df.drop(columns = ['negativereason_gold'], inplace = True)

**Add a column for number of negative reasons.**

In [19]:
# for nan, the value is 0. otherwise the number of reasons
df['no_negative_reasons'] = df['negativereason'].apply(lambda x: 0 if x != x else len(x.split('\n'))) 
df.head()

tweet_id airline_sentiment  airline_sentiment_confidence  \
0  570306133677760513           neutral                        1.0000   
1  570301130888122368          positive                        0.3486   
2  570301083672813571           neutral                        0.6837   
3  570301031407624196          negative                        1.0000   
4  570300817074462722          negative                        1.0000   

  negativereason  negativereason_confidence         airline        name  \
0            NaN                        NaN  Virgin America     cairdin   
1            NaN                     0.0000  Virgin America    jnardino   
2            NaN                        NaN  Virgin America  yvonnalynn   
3     Bad Flight                     0.7033  Virgin America    jnardino   
4     Can't Tell                     1.0000  Virgin America    jnardino   

   retweet_count                                               text  \
0              0                @VirginAmerica What @dhepburn said.   
1              0  @VirginAmerica plus you've added commercials t...   
2              0  @VirginAmerica I didn't today... Must mean I n...   
3              0  @VirginAmerica it's really aggressive to blast...   
4              0  @VirginAmerica and it's a really big bad thing...   

  tweet_coord              tweet_created tweet_location  \
0         NaN  2015-02-24 11:35:52 -0800            NaN   
1         NaN  2015-02-24 11:15:59 -0800            NaN   
2         NaN  2015-02-24 11:15:48 -0800      Lets Play   
3         NaN  2015-02-24 11:15:36 -0800            NaN   
4         NaN  2015-02-24 11:14:45 -0800            NaN   

                user_timezone  no_negative_reasons  
0  Eastern Time (US & Canada)                    0  
1  Pacific Time (US & Canada)                    0  
2  Central Time (US & Canada)                    0  
3  Pacific Time (US & Canada)                    1  
4  Pacific Time (US & Canada)                    1

### Drop `tweet_coord`

In [20]:
df.tweet_coord.value_counts()

[0.0, 0.0]                      164
[40.64656067, -73.78334045]       6
[40.64646912, -73.79133606]       3
[32.91792297, -97.00367737]       3
[39.83426941, -104.69960636]      2
                               ... 
[42.36494288, -71.01451482]       1
[25.05152514, -77.46620117]       1
[29.98498519, -95.3338971]        1
[41.05811177, -73.53637216]       1
[33.94729153, -118.40186744]      1
Name: tweet_coord, Length: 832, dtype: int64

- `tweet_coord` shows the coordinates of the contributors. We could drop it.

In [21]:
df.drop(columns = ['tweet_coord'], inplace = True)

In [22]:
check_missing_values()

# of missing values  % of missing values
negativereason                            5462                37.31
negativereason_confidence                 4118                28.13
tweet_location                            4733                32.33
user_timezone                             4820                32.92

## Drop `tweet_id` and `name`
- `tweet_id` and `name` don't contain much useful information for our analysis. We could drop them.

In [23]:
df.drop(columns = ['tweet_id', 'name'], inplace = True)

## Check unique values

In [24]:
for col in df.columns:
    print(col, ":", df[col].nunique())

airline_sentiment : 3
airline_sentiment_confidence : 1023
negativereason : 18
negativereason_confidence : 1410
airline : 6
retweet_count : 18
text : 14427
tweet_created : 14247
tweet_location : 3081
user_timezone : 85
no_negative_reasons : 3


## Check `tweet_location`

In [25]:
list(df.tweet_location.unique())

[nan,
 'Lets Play',
 'San Francisco CA',
 'Los Angeles',
 'San Diego',
 '1/1 loner squad',
 'NYC',
 'San Francisco, CA',
 'palo alto, ca',
 'west covina',
 'this place called NYC',
 'Somewhere celebrating life. ',
 'Boston | Waltham',
 'Boston, MA ',
 '714',
 'San Mateo, CA & Las Vegas, NV',
 'Brooklyn',
 'California, San Francisco',
 'Washington DC',
 'Texas',
 'Worldwide',
 'Central Texas',
 "i'm creating a monster ",
 'Iowa City',
 'Georgia',
 'Turks and caicos',
 'Oakland via Midwest ',
 'New York, NY',
 'Northern Virginia',
 'Los Angeles / Atlanta',
 'new york, new york',
 'brooklyn, Ny',
 'Bali, Republic of Indonesia',
 'UK, USA. ',
 'Gold Coast, Australia',
 'Stockton, CA',
 'Twin Cities, Minn.',
 'USA',
 'next city',
 'SF ↔ NY',
 'New York + Panama',
 'London, England',
 'Floridian from Cincinnati ',
 'Dallas, Texas',
 'Seattle, WA',
 'Lower Pacific Heights, SF, CA',
 'Chicago',
 'Los Cabos,Mexico',
 'Los Angeles, CA',
 'Austin, TX',
 'Sterling Heights, MI',
 'Manhattan Beach, 

- Representations of `tweet_location` are mixed. Some are real locations, others are not. Granularity of geographic locations is from borough/city to state/country. And they're not limited to English representations.

## Split `tweet_created`

In [26]:
df['tweet_created'] = pd.to_datetime(df['tweet_created'])

In [27]:
df['year'] = df['tweet_created'].dt.year
df['month'] = df['tweet_created'].dt.month
df['day'] = df['tweet_created'].dt.day
df['hour'] = df['tweet_created'].dt.hour

In [28]:
df.head()

airline_sentiment  airline_sentiment_confidence negativereason  \
0           neutral                        1.0000            NaN   
1          positive                        0.3486            NaN   
2           neutral                        0.6837            NaN   
3          negative                        1.0000     Bad Flight   
4          negative                        1.0000     Can't Tell   

   negativereason_confidence         airline  retweet_count  \
0                        NaN  Virgin America              0   
1                     0.0000  Virgin America              0   
2                        NaN  Virgin America              0   
3                     0.7033  Virgin America              0   
4                     1.0000  Virgin America              0   

                                                text  \
0                @VirginAmerica What @dhepburn said.   
1  @VirginAmerica plus you've added commercials t...   
2  @VirginAmerica I didn't today... Must mean I n...   
3  @VirginAmerica it's really aggressive to blast...   
4  @VirginAmerica and it's a really big bad thing...   

              tweet_created tweet_location               user_timezone  \
0 2015-02-24 11:35:52-08:00            NaN  Eastern Time (US & Canada)   
1 2015-02-24 11:15:59-08:00            NaN  Pacific Time (US & Canada)   
2 2015-02-24 11:15:48-08:00      Lets Play  Central Time (US & Canada)   
3 2015-02-24 11:15:36-08:00            NaN  Pacific Time (US & Canada)   
4 2015-02-24 11:14:45-08:00            NaN  Pacific Time (US & Canada)   

   no_negative_reasons  year  month  day  hour  
0                    0  2015      2   24    11  
1                    0  2015      2   24    11  
2                    0  2015      2   24    11  
3                    1  2015      2   24    11  
4                    1  2015      2   24    11

## Data description

In [29]:
df.describe()

airline_sentiment_confidence  negativereason_confidence  retweet_count  \
count                  14640.000000               10522.000000   14640.000000   
mean                       0.900169                   0.638298       0.082650   
std                        0.162830                   0.330440       0.745778   
min                        0.335000                   0.000000       0.000000   
25%                        0.692300                   0.360600       0.000000   
50%                        1.000000                   0.670600       0.000000   
75%                        1.000000                   1.000000       0.000000   
max                        1.000000                   1.000000      44.000000   

       no_negative_reasons     year    month           day          hour  
count         14640.000000  14640.0  14640.0  14640.000000  14640.000000  
mean              0.627527   2015.0      2.0     20.947473     12.319945  
std               0.484750      0.0      0.0      2.170646      5.306285  
min               0.000000   2015.0      2.0     16.000000      0.000000  
25%               0.000000   2015.0      2.0     19.000000      8.000000  
50%               1.000000   2015.0      2.0     22.000000     12.000000  
75%               1.000000   2015.0      2.0     23.000000     17.000000  
max               2.000000   2015.0      2.0     24.000000     23.000000

In [30]:
df.describe(include = ['object'])

airline_sentiment          negativereason airline            text  \
count              14640                    9178   14640           14640   
unique                 3                      18       6           14427   
top             negative  Customer Service Issue  United  @united thanks   
freq                9178                    2908    3822               6   

       tweet_location               user_timezone  
count            9907                        9820  
unique           3081                          85  
top        Boston, MA  Eastern Time (US & Canada)  
freq              157                        3744

- Most of the sentiments are negative.
- Customer Service Issue is the #1 complaint.
- Among six airlines, United appears most frequently.
- About 30% contributors are from Eastern Time (US & Canada) time zone.
- All the data was collected during 2015-2-16 to 2015-2-24.
- Most of the tweets don't get retweeted. But some outliers get as many as 44 retweets.

# EDA
## Univariate Analysis